Module#1

Fat-Frumos opened the pull request.

Fat-Frumos committed:
Added task 1: Branching strategy

Fat-Frumos committed:
Added tasks 1 and 2 of the build tool module

Fat-Frumos committed:
Merge branch 'gradle' into module#1

Fat-Frumos committed:
Added 10 new lines in firstFile

Fat-Frumos committed:
Added 10 new lines in secondFile

Fat-Frumos committed:
Merge branch 'git#1' into git#2

Fat-Frumos committed:
The first 2 lines in secondFile have been updated

Fat-Frumos committed:
The first 2 lines with the different info in secondFile have been updated

Fat-Frumos committed:
The conflict is revolved, 4 lines with the different info in secondFile have been updated

Fat-Frumos committed:
The first 2 lines in firstFile have been updated

Fat-Frumos committed:
The another 2 lines in firstFile have been updated

Fat-Frumos committed:
The two lines in firstFile have been updated

Fat-Frumos committed:
The last 2 lines in firstFile have been updated

Fat-Frumos committed:
Merge branch 'git#1' of https://github.com/Fat-Frumos/Lab into git#1

Fat-Frumos committed:
Added local maven repository
Created fat-jar for Core module
Removed gradle from utils module

Fat-Frumos committed:
Added new file

Fat-Frumos committed:
Added new lines in file

sslichnyi commented on utils/src/main/java/ua/com/numbers/StringUtils.java:
```diff
@@ -0,0 +1,13 @@
+        System.out.println(StringUtils.isPositiveNumber(args[0]));
+    }
+
+    public static boolean isPositiveNumber(String str) {
```
This method should more flexible. 
`isPositiveNumber(String... strs)`

Fat-Frumos replied:
Made a more flexible method 
boolean isPositiveNumbers(String... args)

sslichnyi commented on utils/src/test/java/ua/com/numbers/StringUtilsTest.java:
```diff
@@ -0,0 +1,23 @@
+
+class StringUtilsTest {
+    @Test
+    public void testPositiveNumberReturnFalse() {
```
Please use ParametrizedTest

Links:
- https://www.baeldung.com/parameterized-tests-junit-5

Fat-Frumos replied:
Added parameterized tests using annotations: @ValueSource, @CsvSource, @MethodSource
Added descriptions for tests

sslichnyi reviewed (commented):
Please, change it

Fat-Frumos committed:
Made a refactor
Added parameterized tests
Changed a more flexible method StringUtils

Fat-Frumos committed:
Merge branch 'gradle' into module#1

sslichnyi commented on Module#01/gradle/multi-project/core/src/main/java/com/epam/esm/core/Utils.java:
```diff
@@ -0,0 +1,12 @@
+public class Utils {
+    public static boolean isAllPositiveNumbers(String... array) {
+        return array != null && array.length != 0 &&
+                Arrays.stream(array).allMatch(StringUtils::isPositiveNumber);
```
It is written extremely good with all validation checks.

sslichnyi commented on Module#01/gradle/multi-project/core/src/test/java/com/epam/esm/core/UtilsTest.java:
```diff
@@ -0,0 +1,106 @@
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"1", "2", "3"})
```
Perfect! ParametrizedTest is what I wanted to see here!

Fat-Frumos committed:
Removed nested repository

sslichnyi commented on Module#01/gradle/multi-project/core/build.gradle:
```diff
@@ -0,0 +1,7 @@
+}
+
+dependencies {
+    implementation files('libs/utils-1.3.5.jar')
```
It is implemented in wrong way. You have to use mavenCentral repository. Please, read more about https://stackoverflow.com/questions/35094655/publish-java-artifact-to-maven-local-with-gradle.

You need to add utils-1.3.5.jar as dependency but not as lib file.

Fat-Frumos replied:
Added repositories{ maven local, maven central} in build.gradle

sslichnyi commented:
please, delete 'git' folder

sslichnyi commented on Module#01/gradle/utils/app/src/main/java/com/epam/esm/StringUtils.java:
```diff
@@ -0,0 +1,20 @@
+
+public class StringUtils {
+    public static void main(String[] args) {
+        System.out.println(isPositiveNumbers(args));
```
...Utils class must not contain main() method. StringUtils is more like a library. It is not an application.

Fat-Frumos replied:
The removed the main method in StringUtils.java

sslichnyi commented on Module#01/gradle/utils/app/src/main/java/com/epam/esm/StringUtils.java:
```diff
@@ -0,0 +1,20 @@
+        System.out.println(isPositiveNumbers(args));
+    }
+
+    protected static boolean isPositiveNumbers(String... args) {
```
this method should not be protected. it must be public. This method will be used widely in other applications.

sslichnyi commented on Module#01/gradle/utils/app/build.gradle:
```diff
@@ -0,0 +1,48 @@
+
+repositories {
+    mavenLocal()
+    mavenCentral()
```
It is good. It is exactly what was required.

sslichnyi commented:
There is no info (commands) in README.md file about steps which programmer should do in order to build and publish your utils jar dependency into local maven repository. Please, provide it.

sslichnyi commented:
I didn't find any README.md or any other doc, where git task was documented.

sslichnyi reviewed (approved):
Please, fix found issues

Fat-Frumos committed:
Merged changes from git#2

Fat-Frumos committed:
Merge branch 'git#1' into module#1

Fat-Frumos committed:
Merge branch 'git#3' into module#1

Fat-Frumos committed:
Removed folder git
Added gradle commands to README.ms for building, testing, running, publishing
Added git commands to root module#01 README.md for git task
Added to build.gradle repositories{ maven local, maven central}
Changed visibility to method isPositive Number, removed main method in StringUtils
Created java doc, package info, removed git-action file
